docs(guide/Migrating from Previous Versions): add info for 1.4 ngPatt…

diegomrsantos opened the pull request:
…ern and pattern breaking change

Introduced in commit 0e00108

Closes #15758

diegomrsantos committed:
docs(guide/Migrating from Previous Versions): add info for 1.4 ngPattern and pattern breaking change

Introduced in commit 0e00108

Closes #15758

diegomrsantos commented:
I signed it!

Narretz commented on docs/content/guide/migration.ngdoc:
```diff
@@ -1683,6 +1683,79 @@ ngModelCtrl.$formatters.push(function(value) {
+
+Due to [0e001084](https://github.com/angular/angular.js/commit/0e001084ffff8674efad289d37cb16cc4e46b50a),
+The `ngPattern` and `pattern` directives will validate the regex
+against the `viewValue` of `ngModel`, i.e. the value of the model
```
You should prefix viewValue, modelValue with `$` as this is how they are stored in the model

Narretz commented:
Can you add a note to the commit message that the content in the migration guide is taken from the commit message of 0e00108?

diegomrsantos committed:
docs(guide/Migrating from Previous Versions): prefix viewValue and modelValue with $ as this is how they are stored in the model

This content being included in the migration guide is taken from the commit message of 0e00108
